esp_pixel_driver.ino

```diff
@@ -140,24 +170,75 @@ void status(uint8_t state)
       break;
     }
 
-    if (OUTPUT_MODE == OUTPUT_MODE_LED)
-    {
+    if (config.output_leds) {
       leds[i] = CRGB(r, g, b);
     }
     debug_led_output(i, r, g, b);
   }
-  if (OUTPUT_MODE == OUTPUT_MODE_LED)
-  {
+  if (config.output_leds) {
     delay(500);
     FastLED.show();
   }
 }
 
-void debug_led_output(int i, uint8_t r, uint8_t g, uint8_t b)
-{
-  if (LOG_LEVEL == LOG_LEVEL_DEBUG)
-  {
-    char buf[16]; //formatting buffer
+void onDmxFrame(uint16_t universe, uint16_t length, uint8_t sequence,
+                uint8_t *data) {
+  DmxParameters params = DmxParameters(universe, length, sequence, data, leds);
+  TaskHandle_t dxmHandle = NULL;
+  if (xQueueSend(dmxQueue, (void *)&params, (TickType_t)10) != pdPASS) {
+    /* Failed to post the message, even after 10 ticks. */
+  }
+}
+
+void dmxTask(void *parameters) {
+  bool sendFrame = 1;
+  for (;;) {
+    if (dmxQueue != NULL) {
+      DmxParameters params;
+      if (xQueueReceive(dmxQueue, &(params), (TickType_t)10) == pdPASS) {
+        sendFrame = 1;
+
+        // Mark received Universe
+        if ((params.universe - config.start_universe) < maxUniverses) {
+          universesReceived[params.universe - config.start_universe] = 1;
+        }
+
+        for (int i = 0; i < maxUniverses; i++) {
+          if (universesReceived[i] == 0) {
+            // if we're missing an expected universe, do not playback
+            sendFrame = 0;
+            break;
+          }
+        }
+
+        // read universe and put into the right part of the display buffer
```

Comment:
i feel like when inline comments seem necessary its a good sign that the logic should be broken out into a function.